### PR TITLE
optimize imageMgaick commands and doDiff function to improve speed

### PR DESCRIPTION
**Pull request by freakin09**

### CODE REVIEW

File: src/imageMagickDiff.js [hunk 1]
```diff
@@ -1,23 +1,19 @@
 import child_process from 'child_process';
 import path from 'path';
+import fs from 'fs'
 
 var DimensionOfImage;
 
 //TODO: for very large images, perhaps split it into smaller pieces and
 // then compare to optimize ?
-var doDiff = async function(filename1, filename2, diffFile,
-                            includeOnlyBoxes, skipBoxes) {
```
[the remark this answers is not in the document]
Comment (freakin09) [reply src/imageMagickDiff.js]: includeOnly is faster using the older method as it only compares the includeOnly pixels...so in phantesta.js, if there is any includeOnlyBox, phantesta comapres using the older method

[the remark this answers is not in the document]
Comment (freakin09) [reply src/imageMagickDiff.js]: yes..backwards compatible

File: src/phantesta.js [hunk 2]
```diff
@@ -301,11 +318,40 @@ Phantesta.prototype.expectDiff = async function(name1, name2, excludeBoxes, incl
         'success: ' + name1 + ' is different than ' + name2);
   }
 };
-Phantesta.prototype.isDiff = function(filename1, filename2, diffPath, excludeBoxes, includeBoxes) {
+Phantesta.prototype.isDiff = async function(filename1, filename2, diffPath, excludeBoxes, includeBoxes) {
+
+  if(!includeBoxes || !includeBoxes.length) {
```
Comment (stoarca) [inline src/phantesta.js]: space after if

File: src/phantesta.js [hunk 3]
```diff
@@ -301,11 +318,40 @@ Phantesta.prototype.expectDiff = async function(name1, name2, excludeBoxes, incl
...
+    return window.imageDiffer
+        .includeOnlyBoxes(arguments[1])
+        .censorBoxes(arguments[0])
+        .doDiff();
+  }, excludeBoxes, includeBoxes);
+  var ret = null;
+  while (ret === null || ret === 'RESULT NOT READY') {
+    ret = await this.diffPage.evaluate(function() {
+      if (!window.imageDiffer.isReady()) {
+        return 'RESULT NOT READY';}
```
Comment (stoarca) [inline src/phantesta.js]: closing brace on new line

**Pull request merged**